lib/components/app/trip-preview-layout-base.tsx

```diff
@@ -0,0 +1,138 @@
+import { Button } from 'react-bootstrap'
+import { connect } from 'react-redux'
+import { FormattedMessage } from 'react-intl'
+import { Itinerary } from '@opentripplanner/types'
+import { Map } from '@styled-icons/fa-solid/Map'
+import { Print } from '@styled-icons/fa-solid/Print'
+import { Times } from '@styled-icons/fa-solid/Times'
+// @ts-expect-error not typescripted yet
+import PrintableItinerary from '@opentripplanner/printable-itinerary'
+import React, { Component, ReactNode } from 'react'
+
+import {
+  addPrintViewClassToRootHtml,
+  clearClassFromRootHtml
+} from '../../util/print'
+import { AppConfig } from '../../util/config-types'
+import { AppReduxState } from '../../util/state-types'
+import { ComponentContext } from '../../util/contexts'
+import { IconWithText } from '../util/styledIcon'
+import PageTitle from '../util/page-title'
+import SpanWithSpace from '../util/span-with-space'
+import TripDetails from '../narrative/connected-trip-details'
+
+type Props = {
+  config: AppConfig
+  header?: ReactNode
+  itinerary?: Itinerary
+  mapElement?: ReactNode
+  onClose?: () => void
+  subTitle?: string
+  title: string
+}
+
+type State = {
+  mapVisible?: boolean
+}
+
+class TripPreviewLayoutBase extends Component<Props, State> {
```

Comment:
What was the reason behind going for a class component here rather than a function component?

Reply:
Copy/paste is the reason...